extend user profiles and fixup application layout

xrmx opened the pull request:
More details in single commits

xrmx committed:
models: rename User as UserProfile

As the the users should be able to login we need a fk to django
contrib.auth's User.

xrmx committed:
models: add a few more field to the userprofile

Add address, bio, picture, tax code and vat_id optional fields.

xrmx committed:
README: update instructions

To match the one relevant for an app and not for a whole django
project.

xrmx committed:
urls: don't leak views into the project urls

Instead import the urls from labAdmin app

xrmx committed:
tests: rework tests to something that resembles a test

Even if there's not tests actually :|

xrmx committed:
Use proper django app layout

So that it is installable and the test project can run tests

xrmx committed:
models: fix typos

xrmx committed:
models: signup dates should be nullable

As we should not need them when we create users

xrmx committed:
labAdmin: add missing migrations

xrmx committed:
utils: add fablabto specific utils

One script to scrape the current plone server and another
one to import dumped data here.

mastrolinux commented on setup.py:
```diff
@@ -0,0 +1,43 @@
+    description="""A django app to manage your Fablab""",
+    author='Fablab Torino',
+    author_email='info@fablabtorino.org',
+    url='https://github.com//FablabTorino/LabAdmin',
```
there is an extra slash here.

xrmx replied:
fixed, thanks

xrmx committed:
setup: fix project url

xrmx committed:
Hook oauth2 server machinery in the test project

xrmx committed:
labAdmin: Add crude login template

We need to hook login url that does not ship with a template so if
the project where we are installed does not have one provide
ourselves.
Hook django login and logout views to our testSite so we can
test the oauth2 integration.

xrmx committed:
labAdmin: add view to serve user data from an oauth token

This is neede for rocket.chat integration.

While at it update README and setup with the django-oauth-toolkit
dependency.

xrmx committed:
labAdmin: add link to reset password to login template

And hook reset views on the test project

mastrolinux committed:
Merge pull request #13 from FablabTorino/oauth2

Hook oauth2 server machinery in the test project
